src/ParameterDistributions.jl

```diff
@@ -873,87 +839,51 @@ function transform_constrained_to_unconstrained(
-"""
-function transform_unconstrained_to_constrained(
-    pd::ParameterDistribution,
-    x::AbstractVector{FT};
-    build_flag::Bool = true,
-) where {FT <: Real}
-    param_names = get_name(pd)
-    pd_constraints = get_all_constraints(pd, return_dict = true)
+Apply the transformation to map unconstrained parameter sample(s) `x` into the constrained space.
 
-    # naive function parameter check, is x a dof vector, or the unconstrained evaluated function?
-    if build_flag
-        pd_batch_idxs = batch(pd, function_parameter_opt = "dof")
-    else
-        pd_batch_idxs = batch(pd, function_parameter_opt = "eval")
-    end
-
-    ret = []
-    for (name, idxs, d) in zip(param_names, pd_batch_idxs, pd.distribution)
-        push!(
-            ret,
-            Array(transform_unconstrained_to_constrained(d, pd_constraints[name], x[idxs], build_flag = build_flag)),
-        )
-    end #returns a list of params (though only 1 sample) that are concatenated 
-
-    return cat(ret..., dims = 1)
-
-end
-
-"""
-    transform_unconstrained_to_constrained(d::ParameterDistribution, x::AbstractMatrix)
-
-Apply the transformation to map unconstrained parameter samples `x` (samples stored as columns) into the constrained space.
+Each column of `x` is a sample, and each row is a parameter.
 """
 function transform_unconstrained_to_constrained(
     pd::ParameterDistribution,
-    x::AbstractMatrix{FT};
+    x::AbstractArray{FT};
     build_flag::Bool = true,
 ) where {FT <: Real}
     param_names = get_name(pd)
-    pd_dists = get_distribution(pd)
     pd_constraints = get_all_constraints(pd, return_dict = true)
 
     # naive function parameter check, is x a dof vector, or the unconstrained evaluated function?
-    if build_flag
-        pd_batch_idxs = batch(pd, function_parameter_opt = "dof")
+    function_parameter_opt = build_flag ? "dof" : "eval"
+    pd_batch_idxs = batch(pd; function_parameter_opt)
+
+    x_out_size, eval_batch_idxs = if build_flag && (eltype(pd.distribution) == GaussianRandomFieldInterface)
+        (sum(n_eval_pts.(pd.distribution)), length(axes(x, 2))), batch(pd; function_parameter_opt = "eval")
     else
-        pd_batch_idxs = batch(pd, function_parameter_opt = "eval")
+        size(x), pd_batch_idxs
     end
-
-    ret = []
-    for (name, idxs, d) in zip(param_names, pd_batch_idxs, pd.distribution)
-        push!(
-            ret,
-            Array(transform_unconstrained_to_constrained(d, pd_constraints[name], x[idxs, :], build_flag = build_flag)),
-        )
-    end #returns a list of row-blocks that are concatenated
-
-    return cat(ret..., dims = 1)
+    x_out = similar(x, x_out_size)
+    for (name, eval_idx, pd_idxs, d) in zip(param_names, eval_batch_idxs, pd_batch_idxs, pd.distribution)
+        view(x_out, eval_idx, :) .=
+            transform_unconstrained_to_constrained(d, pd_constraints[name], view(x, pd_idxs, :); build_flag)
+    end
+    return x_out
 
 end
```

Comment:
Again, much of the discussion is the same as for `transform_constrained_to_unconstrained`, above.
There is one added complication however: The `dof` batch is different from the `eval` batch. In other words, the elements of `x` _may_ be interpreted as coefficients of a certain function as opposed to evaluation points. As such, we need to fetch the correct output size (`x_out_size`) and indices (`eval_batch_idxs`) that match the _evaluation_ size.

Reply:
again I would rather we call ndims with kwarg `evals` than call `n_eval_pts` directly

Reply:
done.